### PR TITLE
add jitter to timer / transfer / replication queue sanity scan

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: service/history/queueAckMgr_test.go [hunk 1]
```diff
@@ -140,7 +140,10 @@ func (s *queueAckMgrSuite) SetupTest() {
 	}
 	s.mockShard.config.ShardUpdateMinInterval = dynamicconfig.GetDurationPropertyFn(0 * time.Second)
 
-	s.queueAckMgr = newQueueAckMgr(s.mockShard, &QueueProcessorOptions{UpdateShardTaskCount: 1, MetricScope: metrics.ReplicatorQueueProcessorScope}, s.mockProcessor, 0, s.logger)
+	s.queueAckMgr = newQueueAckMgr(s.mockShard, &QueueProcessorOptions{
+		UpdateShardTaskCount: func(opts ...dynamicconfig.FilterOption) int { return 1 },
```
Comment (vancexu) [inline service/history/queueAckMgr_test.go]: Could use dynamicconfig.GetIntPropertyFn() which is mock for tests.

File: service/history/queueAckMgr_test.go [hunk 2]
```diff
@@ -346,7 +349,10 @@ func (s *queueFailoverAckMgrSuite) SetupTest() {
 	}
 	s.mockShard.config.ShardUpdateMinInterval = dynamicconfig.GetDurationPropertyFn(0 * time.Second)
 
-	s.queueFailoverAckMgr = newQueueFailoverAckMgr(s.mockShard, &QueueProcessorOptions{UpdateShardTaskCount: 1, MetricScope: metrics.ReplicatorQueueProcessorScope}, s.mockProcessor, 0, s.logger)
+	s.queueFailoverAckMgr = newQueueFailoverAckMgr(s.mockShard, &QueueProcessorOptions{
+		UpdateShardTaskCount: func(opts ...dynamicconfig.FilterOption) int { return 1 },
```
Comment (vancexu) [inline service/history/queueAckMgr_test.go]: Could use dynamicconfig.GetIntPropertyFn()

File: common/service/dynamicconfig/constants.go [hunk 3]
```diff
@@ -177,6 +177,8 @@ const (
 	TimerProcessorCompleteTimerInterval
 	// TimerProcessorMaxPollInterval is max poll interval for timer processor
 	TimerProcessorMaxPollInterval
+	// TimerProcessorMaxPollIntervalJitterCoefficient is the max poll interval jitter coefficient
+	TimerProcessorMaxPollIntervalJitterCoefficient
```
Comment (vancexu) [inline common/service/dynamicconfig/constants.go]: There new added keys also need to updated in `keys` map in this file.

File: service/history/queueProcessor.go [hunk 4]
```diff
@@ -156,10 +160,10 @@ processorPumpLoop:
 			p.processBatch(tasksCh)
 		case <-pollTimer.C:
 			p.processBatch(tasksCh)
-			pollTimer.Reset(p.options.MaxPollInterval)
+			pollTimer.Reset(jitter.JitDuration(p.options.MaxPollInterval(), p.options.MaxPollIntervalJitterCoefficient()))
```
Comment (samarabbas) [inline service/history/queueProcessor.go]: I think we want same logic to skip the forced scan if already performed recently.

**Pull request merged**